File: openclaw-android/app/src/main/java/de/openclaw/assistant/data/model/ChatMessage.kt
package de.openclaw.assistant.data.model

data class ChatMessage(
    val id: String,
    val role: String,      // "user" | "assistant"
    val content: String,
    val model: String? = null,
    val timestamp: Long = System.currentTimeMillis()
)

data class ChatRequest(
    val message: String,
    val model: String = "gemini-2.5-flash",
    val stream: Boolean = false
)

data class ChatResponse(
    val id: String,
    val content: String,
    val model: String,
    val tokens: Int
)

data class AuthRequest(
    val email: String,
    val password: String
)

data class AuthResponse(
    val accessToken: String,
    val refreshToken: String,
    val user: UserProfile
)

data class UserProfile(
    val id: String,
    val email: String,
    val displayName: String,
    val tier: String    // "FREE" | "PREMIUM"
)

File: openclaw-ios/OpenClaw/Models/ChatMessage.swift
import Foundation

struct ChatMessage: Identifiable, Codable {
    let id: String
    let role: String          // "user" | "assistant"
    let content: String
    let model: String?
    let timestamp: Date

    init(id: String = UUID().uuidString,
         role: String, content: String,
         model: String? = nil,
         timestamp: Date = Date()) {
        self.id = id
        self.role = role
        self.content = content
        self.model = model
        self.timestamp = timestamp
    }
}

struct ChatRequest: Codable {
    let message: String
    let model: String
    let stream: Bool

    init(message: String,
         model: String = "gemini-2.5-flash",
         stream: Bool = false) {
        self.message = message
        self.model = model
        self.stream = stream
    }
}

struct ChatResponse: Codable {
    let id: String
    let content: String
    let model: String
    let tokens: Int
}

struct AuthRequest: Codable {
    let email: String
    let password: String
}

struct AuthResponse: Codable {
    let accessToken: String
    let refreshToken: String
    let user: UserProfile
}

struct UserProfile: Codable {
    let id: String
    let email: String
    let displayName: String
    let tier: String  // "FREE" | "PREMIUM"
}
